src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -182,6 +184,15 @@ public static Map<String, Object> parseStringToObjectMap(XContentParser parser)
             if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
                 // If the current token is a START_OBJECT, parse it as Map<String, String>
                 map.put(fieldName, parseStringToStringMap(parser));
+            } else if (parser.currentToken() == XContentParser.Token.START_ARRAY) {
+                // If an array, parse it to a string
+                // Handle array: convert it to a string representation
+                List<String> elements = new ArrayList<>();
+                while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                    elements.add("\"" + parser.text() + "\"");  // Adding escaped quotes around each element
```

Comment:
how the parser handles special characters? could there be a situation we need additional escaping for some characters?

Reply:
The parser can deal with strings like, haven't tried this within an array but also that isn't an allowed input in downstream apis : 
```
 "\n    StringBuilder parametersBuilder = new StringBuilder(\"{\");\n    if (params.text_docs.length > 0 && params.text_docs[0] != null) {\n      parametersBuilder.append(\"\\\"inputText\\\":\");\n      parametersBuilder.append(\"\\\"\");\n      parametersBuilder.append(params.text_docs[0]);\n      parametersBuilder.append(\"\\\"\");\n      \n      if (params.text_docs.length > 1 && params.text_docs[1] != null) {\n        parametersBuilder.append(\",\");\n      }\n    }\n    \n    \n    if (params.text_docs.length > 1 && params.text_docs[1] != null) {\n      parametersBuilder.append(\"\\\"inputImage\\\":\");\n      parametersBuilder.append(\"\\\"\");\n      parametersBuilder.append(params.text_docs[1]);\n      parametersBuilder.append(\"\\\"\");\n    }\n    parametersBuilder.append(\"}\");\n    \n    return  \"{\" +\"\\\"parameters\\\":\" + parametersBuilder + \"}\";",
 ```

Reply:
tbh this was a good call out cause, I am not sure if my fix somewhere else broke taking arrays of numbers but now code will be: 
```else if (parser.currentToken() == XContentParser.Token.START_ARRAY) {
                // If an array, parse it to a string
                // Handle array: convert it to a string representation
                List<String> elements = new ArrayList<>();
                while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
                    if (parser.currentToken().equals(XContentParser.Token.VALUE_NUMBER)) {
                        elements.add(String.valueOf(parser.numberValue()));  // If number value don't add escaping quotes
                    } else {
                        elements.add("\"" + parser.text() + "\"");  // Adding escaped quotes around each element
                    }
                }
                map.put(fieldName, elements.toString());
            }
            ```

Reply:
@jackiehanyang @owaiskazi19 lmk what you think of above tested it with different inputs and all templates

Reply:
I'm still not very confident in manually wrapping `"\"` around elements for escape purpose. For example, it will cause incorrect escape result for strings start with quotes and control-chars like backslash. But it could a rare edge case, I'm fine with revising this in a separate pr/issue.

Reply:
Will create an issue for this